### ISSUE TITLE
The shortcut to show the terminal doesn't work on windows

### ISSUE THREAD

**Issue by shakesBeardZ** [microsoft/vscode#71101]
after a whole year of using vs code, I always find that
the shortcut to show terminal doesn't work on windows (Ctrl+`)
is there any solution

**Comment by Tyriar** [comment]
Can you open the terminal through the menu? What keyboard layout are you using? Can you post a screenshot of this menu entry:

![image](https://user-images.githubusercontent.com/2193314/54944693-557f4600-4ef1-11e9-9585-46dce5ee2fa9.png)

**Comment by shakesBeardZ** [comment]
@Tyriar  yes I can open it by clicking on the menu but the shortcut ``` Ctrl+` ```  is not working 

![image](https://user-images.githubusercontent.com/15571628/54947915-bf244200-4f3b-11e9-92a7-ed462d339a24.png)

**Comment by alexandrudima** [comment]
@shakesBeardZ Is VS Code receiving the keybinding from the OS?

![keyboard-input](https://cloud.githubusercontent.com/assets/5047891/25478333/fe31c830-2b40-11e7-865c-b2c3bb10dd75.gif)

**Comment by klick-barakgall** [comment]
@alexandrudima I'm experiencing the same issue, when I try to create a keyboard shortcut pressing ``ctrl+` `` it instead gives me `ctrl+unknown`..

I recently changed my keyboard input scheme. I default to the Microsft IME input keyboard, which I guess gives a different code for backtick?

However when I switch to the default US keyboard, it starts working, but the shortcut dialog records it as `ctrl+oem_3`.

Any chance we can get that cleaned up and also maybe support for some other input keyboards?

**Comment by alexandrudima** [comment]
@klick-barakgall What keyboard layout have you configured on Windows?

**Comment by klick-barakgall** [comment]
@alexandrudima I have it set to the default `English keyboard (101/102 key)` layout. My only other option being `Japanese keyboard (106/109 key)`. I dunno if that answers your question? If you have a more specific question I'm happy to help.

**Comment by alexandrudima** [comment]
@klick-barakgall I would like to try to reproduce your problem. In order to do that, I need to install and configure the same keyboard layout as you have on your machine. So are you using English/US ?

(Rendered next to the clock like this:)
![image](https://user-images.githubusercontent.com/5047891/55462634-82b9ab80-55f7-11e9-86af-40b1f196a680.png)

(Rendered in the settings like this):
![image](https://user-images.githubusercontent.com/5047891/55462679-96fda880-55f7-11e9-8d7f-1d3f871e8587.png)

or some other keyboard layout? If you have another keyboard layout, can you please take a screenshot of the one you are using so I can try to install the same.

**Comment by klick-barakgall** [comment]
Apologies for not being clearer.

This is my languages configuration:
![image](https://user-images.githubusercontent.com/8572438/55564978-ff678b00-56c6-11e9-8033-7654d8cc6ba7.png)

The Asian language in that list is Japanese FYI.

Within that I have a the keyboards configured as such: 
![image](https://user-images.githubusercontent.com/8572438/55565069-258d2b00-56c7-11e9-9e2d-ee2071edc496.png)

I believe this Japanese keyboard setup is what is messing with keyboard shortcuts, however I also changed the setting so that all applications default to this keyboard:
![image](https://user-images.githubusercontent.com/8572438/55565252-80bf1d80-56c7-11e9-98d4-3ced96ce72ad.png)

Please let me know if that gives you enough information.

**Comment by alexandrudima** [comment]
Thank you for the extra information. Something interesting occurs when pressing `` ctrl + ` `` under the Japanese - Microsoft IME keyboard layout. The received key press is no longer for `Backquote`, but for `Zenkaku`.

* when pressing `Backquote`, we receive it as `Backquote`:

![Kapture 2019-04-29 at 15 44 56](https://user-images.githubusercontent.com/5047891/56900372-f71c2900-6a95-11e9-8b37-b2d7d52b482e.gif)

* when pressing `Ctrl` + `Backquote`, we receive it as `Ctrl` + `Zenkaku`:

![Kapture 2019-04-29 at 15 47 51](https://user-images.githubusercontent.com/5047891/56900464-321e5c80-6a96-11e9-9939-985c06a0252c.gif)

Having briefly searched for `Zenkaku`, I ended up on Wikipedia at https://en.wikipedia.org/wiki/Language_input_keys and it appears it is a dedicated language input key.

In other words, `Ctrl` + `Backquote` is something special under the Japanese IME keyboard layout and I don't think it is expected that applications can bind something to that keypress. I think this is something specific to this keyboard layout and I suggest that you simply rebind the built-in keybinding to something else than `Ctrl` + `Backquote`, e.g. to `Ctrl` + `Shift` + `Backquote`:

```
    {
        "key": "ctrl+shift+`",
        "command": "workbench.action.terminal.toggleTerminal"
    },
```

Would that work for you?

**Comment by klick-barakgall** [comment]
I hate that microsoft doesn't let us remove/change these keybindings.

My preference would rather be to not rebind as I will develop strange habits when I'm trying to help on other VSCode machines. Maybe I can try to find a way to change the default keyboard layout for VSCode instead? Just feels awkward I guess.

I appreciate you looking into it and your continued support.

Actually after further thought, wouldn't it be possible to have multiple keybindings for the same action? If I rebind it to both ``CTRL+`, CTRL+unknown (zenkaku)`` then this solves the issue for me.

I just tried it but I don't see anyway to bind the same action to multiple keybindings right now. Has this become a new feature request?

**Comment by Skriptach** [comment]
Hello everyone

I'm faced same issue today.
I did not change default shortcut or my keyboard.
And when I try to reassign shortcut to ```ctrl+` ```  I get `ctrl+unknown` same as @roa-nyx

**Comment by roa-nyx** [comment]
@Skriptach in my case it ended up being a special microsoft hotkey for the current keyboard layout I had selected (see my screenshots above). Basically I use the Japanese IME keyboard which reserves the ``CTRL+` `` for a keyboard specific action and reports a different keycode for that combination of keys.

In my case I did uncover another VS Code shortcut which does pretty much what I want (it opens a NEW terminal though, which may have slightly different logic, but works for me). If you want give it a shot with: ``CTRL+SHIFT+` ``.

I know this isn't really a resolution, but it's the best we got for now. Alternatively you can still bind the action you want even though it reads `CTRL+unknown` it should still work once you rebind it.

**Comment by alexandrudima** [comment]
To reiterate, ``Ctrl + ` `` is received by VS Code as `CTRL+unknown (zenkaku)` because it is reserved by the Microsoft IME. I cannot comment as to what it does, why the input method doesn't allow it to be configurable, etc... VS Code is just another app running on Windows and receiving key events like any other application...

So this means that ``Ctrl + ` `` is for all intents and purposes unusable.

But all keybindings inside VS Code are configurable 100%. So it is possible to add another keybinding to the "Toggle Integrated Terminal" action. e.g. in `keybindings.json`:
```
{ "key": "ctrl+shift+`", "command": "workbench.action.terminal.toggleTerminal" },
```

There is no code change I can do in VS Code for us to start receiving `` Ctrl + ` ``, so I will close this issue for now, as the workaround is to bind the desired action to another key combination...
